### ISSUE TITLE
Client.parse_response can handle dict's type doesn't allow it.

### ISSUE THREAD

**Issue by Hultner** [OpenIDC/pyoidc#789]
## Background
I'm using `pyoidc` in a ASGI (starlette-based) application. When using a framework like starlette or based upon it the query param's are automatically decoded into dicts.

For my client callback I followed the documentation reference implementation for a code flow. And when I construct my `AuthorizationResponse` I need to encode my data back into json (or urlencoded) for the library to parse it again via `client.parse_response`, causing an unnecessary extra encode/decode roundtrip. 

## Current behaviour
I read the source code and it became apparent that the code technically supports me to provide a dict directly, this is however disallowed by the types both for `sformat: ENCODINGS` where `ENCODINGS = Literal["json", "urlencoded"]` and `info: str`. 

`Client.parse_response` can technically handle dict's but it's unsupported by its type. This can be discovered by looking further into the implementation of `Message.deserialize`, which uses `func = getattr(self, "from_%s" % method)` where `method` originates from `sformat`. Since a `Message.from_dict` exists there's no technical limitation to supporting it from my point of view. I've tried this and it appears to function properly.

## Example code
Currently supported way of doing it (with unnecessary json roundtrip).
```python
auth_resp = client.parse_response(
    AuthorizationResponse,
    info=json.dumps({"code": code, "state": state, "code": code}),
)
```

Works but isn't allowed by the types, hence relies on implementation details not part of the API.
```python
auth_resp = client.parse_response(
    AuthorizationResponse,
    # Can we do this without json dump?
    # info=json.dumps({"code": code, "state": state, "code": code}),
    # Yes we can but it's not officially supported and relies on implementation details.
    # This could break in future versions so it might be worth the extra json round-trip.
    sformat="dict",
    info={"code": code, "state": state, "code": code},
)
```

## Suggestion
Change the type signature of `Client.parse_response` and officially support input via a dictionary.
 
```diff
class Client(PBase):
    ...

    def parse_response(
        self,
        response: Type[Message],
-       info: str = "",
+       info: Union[str, dict] = "",
        # Or make the change in ENCODINGS directly if it work's everywhere.
-       sformat: ENCODINGS = "json",
+       sformat: Union[ENCODINGS, Literal['dict']] = "json",
        state: str = "",
        **kwargs
    ) -> Message:
```

If this is not suitable for other reasons I completely understand but otherwise this would make a lot of sense from my point of view. I can supply a patch/PR if you're willing to incorporate this change.

**Comment by schlenk** [comment]
As the Message base class supports dict, urlencode and json out of the box anyway, this should not be a problem. 
The only possible issue with dicts could be duplicate keys, but i am not aware of any cases where this can happen decoding a message.

From a cursory look, i would say adding ``dict`` directly to the ENCODINGS should work.

So please provide a PR.

**Comment by Hultner** [comment]
@schlenk I added a PR here: #790 790

**Comment by schlenk** [comment]
Closed via #790
